homeassistant/components/zwave_js/update.py

```diff
@@ -89,46 +89,60 @@ def __init__(self, driver: Driver, node: ZwaveNode, semaphore: Semaphore) -> Non
         # device may not be precreated in main handler yet
         self._attr_device_info = get_device_info(driver, node)
 
+    @callback
     def _update_on_status_change(self, _: dict[str, Any]) -> None:
         """Update the entity when node is awake."""
         self._status_unsub = None
         self.hass.async_create_task(self._async_update())
 
-    async def _async_update(self, _: HAEvent | datetime | None = None) -> None:
+    async def _async_update(self, _: HomeAssistant | datetime | None = None) -> None:
         """Update the entity."""
+        self._poll_unsub = None
         for status, event_name in (
             (NodeStatus.ASLEEP, "wake up"),
             (NodeStatus.DEAD, "alive"),
         ):
             if self.node.status == status:
-                if self._status_unsub:
-                    return
-                self._status_unsub = self.node.once(
-                    event_name, self._update_on_status_change
-                )
+                if not self._status_unsub:
+                    self._status_unsub = self.node.once(
+                        event_name, self._update_on_status_change
+                    )
+                return
 
         async with self.semaphore:
             if available_firmware_updates := (
                 await self.driver.controller.async_get_available_firmware_updates(
                     self.node, API_KEY_FIRMWARE_UPDATE_SERVICE
                 )
             ):
-                self._latest_version_firmware = max(
-                    available_firmware_updates,
-                    key=lambda x: AwesomeVersion(x.version),
-                )
+                try:
+                    self._latest_version_firmware = max(
```

Comment:
This shouldn't raise.

Reply:
yep realizing that as I add a test. Fixing